enhancements/installer/storage-class-encrypted.md

```diff
@@ -0,0 +1,350 @@
+---
+title: encrypted-default-storage-class
+authors:
+  - "@patrickdillon"
+reviewers:
+  - "@jhixson74, installer"
+  - "@jsafrane, storage"
+  - "@bertinatto, storage"
+  - "@trilokgeer, storage"
+approvers:
+  - "@jsafrane"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "@joelspeed"
+creation-date: 2022-06-22
+last-updated: 2022-12-09
+tracking-link:
+  - https://issues.redhat.com/browse/CORS-2049
+  - https://issues.redhat.com/browse/OCPPLAN-9307
+---
+
+# Default Storage Class with Encrypted Keys 
+
+## Summary
+
+When user-managed encryption keys are provided at install time, the default storage
+class should be set to utilize those keys. Presently, only root volumes for nodes are
+encrypted with those keys--but users would expect the persistent storage volumes to
+be encrypted as well. This enhancement proposes a solution for AWS, GCP, & Azure.
+
+## Motivation
+
+The primary motivation is to enable managed services to encrypt storage volumes with
+customer-provided keys automatically as part of the installation process without requiring
+day-2 operations. The behavior of automatically setting the default storage class is
+sensible for individual (non-managed-service) users as well; setting the default will
+reduce their day-2 burden and decrease the opportunity for user error.
+
+### User Stories
+
+When I install a cluster, I want to provide user-managed encryption keys in the install-config
+so that the default storage class will be set to provision volumes with those keys.
+
+### Goals
+
+* Default storage class is set to use user-provided keys at cluster creation for public clouds (AWS, GCP, Azure)
+
+### Non-Goals
+* Other clouds besides AWS, GCP, & Azure are out of scope
+
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster creator specifies (aws|gcp|azure) user-managed encryption keys 
+in `defaultMachinePlatform` of install config.
+2. Installer adds encryption key fields to cluster infrastructure object.
+3. Cluster-storage-operator/CSI drivers use cluster infrastructure object
+to create default storage class using user-encrypted keys from infra object.
+
+
+### API Extensions
+
+This enhancement extends the cluster `Infrastructure` API object to add fields
+for the user-managed encryption keys. The fields will be added to the platform
+spec and status for each platform. They are added to the spec so that users can edit the
+spec day 2 in order to update the default storage class. The status field would
+be updated by the cluster-storage operator/CSI driver.
+
+#### AWS 
+
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+	// [... existing fields ...]
+
+    // The KMS key to be used in the default storage class to encrypt volumes.
```

Comment:
Given this is planned to be used for storage, should we include storage in the name or put a new struct in to differentiate. I'm thinking basically, what if in the future we want a KMS key for the secret backend, we would need a new field here and that name might clash

Reply:
This is an interesting comment. Since AWSPlatformSpec can be changed after installation, should be the new key propagated to Cloud API to create new VMs with the new API too? Or is it really just the default storage encryption key?

Reply:
+1 that this is a good point. It is just the default storage encryption key. VM disks are based on the machinesets.

Reply:
MachineSets have their own KMS stuff within them so no need to lean on anything from the infrastructure object right now, as of now, this looks like it is storage only

Reply:
If it's only for storage, does it need to be a field in Infrastructure then? We could also add the fileds to ClusterCSIDriver and install could create that instead.